### PR TITLE
feat: add editor in next.js

### LINKED ISSUE #684: Minimal studio App with a working editor and AsyncAPI documentation
As part of the discussion in #661, I'm creating this issue to track the migration since it's a huge task.

Related to #661

### PR DESCRIPTION
**Pull request by Shurtu-gal**
**Description**

- Add editor to the studio-next in next.js
- Using code mirror instead of Monaco
- Is a minimal editor with file state and local storage baked into it 
- Use `esc + tab` for the normal tab and just `tab` if you want in the editor.

**Related issue(s)**
Fixes #684

### CODE REVIEW

File: apps/studio-next/src/components/Editor/CodeMirror.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+export const CodeMirror = (props: ICodeMirrorProps) => {
+  const { language, value, onChange, autoFocus, className } = props;
+
+  const editorRef = useRef<HTMLDivElement>(null);
+  const editorViewRef = useRef<EditorView>();
+
+  useEffect(() => {
+    let currentValue;
+    if (window) {
+      console.log('Getting asyncapi from local storage...');
```
Comment (Amzani) [inline apps/studio-next/src/components/Editor/CodeMirror.tsx:34]: `console.log` should be removed

**Review by Amzani — APPROVED** [review]
Great PR! @Shurtu-gal 🚀 
I think we can merge it, however, it doesn't completely fix https://github.com/asyncapi/studio/issues/684 as we expect to have a minimal working app with menu + preview documentation.
We will need this as well https://github.com/asyncapi/studio/issues/703

**Comment by Amzani** [comment]
@Shurtu-gal we should start deploying the next-js app in Netlify at least for testing purposes
@fmvilas can you help I guess, I don't have access to Netlify

File: apps/studio-next/src/components/Editor/CodeMirror.tsx (hunk 1 above)
Comment (Shurtu-gal) [reply apps/studio-next/src/components/Editor/CodeMirror.tsx:34]: Sure

**Comment by Shurtu-gal** [comment]
@Amzani should I add those features in this PR or make a separate one.

**Comment by Shurtu-gal** [comment]
> @Shurtu-gal we should start deploying the next-js app in Netlify at least for testing purposes @fmvilas can you help I guess, I don't have access to Netlify

Agreed would be much better as some things might break over there if not reviewed properly.

**Review by Amzani — APPROVED** [review]
LGTM

**Pull request merged**